playbooks/roles/common/tasks/detect-public-ip.yml

```diff
@@ -0,0 +1,51 @@
+---
+# detect-public-ip.yml will attempt to identify whether the server's public
+# IP address is different from what is visible on the host and, if
+# successfully detected, ask to update the address for documentation and
+# configuration profiles
+- name: "Install dns module"
```

Comment:
Ansible has [the `ipify_facts` module](https://docs.ansible.com/ansible/latest/modules/ipify_facts_module.html), but it has a flaw: it returns IPv6 addresses when available.

Consider using https://v4.ident.me/.json via [the `uri` module](https://docs.ansible.com/ansible/latest/modules/uri_module.html).

Reply:
(BTW, this was just the first service I found; I have no special preference for it.)